### PR TITLE
Python sdk synapse Authentication sample

### PR DESCRIPTION
**Pull request by ohadbitt**
#### Pull Request Description

_[Add a description of your pull request here]_

---

#### Future Release Comment
_[Add description of your change, to include in the next release]_
_[Delete any or all irrelevant sections, e.g. if your change does not warrant a release comment at all]_

**Breaking Changes:**
- None

**Features:**
- None

**Fixes:**
- None

### CODE REVIEW

File: samples/src/main/python/SynapsePythonSDK.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# This sample shows how to use the Kusto Python SDK with Synapse authentication.
+# One should use the SDK to run admin commands while using the connector to query
+# https://docs.microsoft.com/azure/data-explorer/kusto/api/python/kusto-python-client-library
+# To upload SDK package to synapse workspace, create a wheel file and follow this:
+# https://docs.microsoft.com/azure/synapse-analytics/spark/apache-spark-manage-python-packages#install-wheel-files
+from azure.kusto.data import KustoClient, KustoConnectionStringBuilder
+
+token_provider = lambda: TokenLibrary().getAccessToken("{\"audience\": \"AzureDataExplorer\", \"name\": \"\"}").getToken()
```
Comment (noam-shapira) [inline samples/src/main/python/SynapsePythonSDK.py:8]: this use user token in interactive mode and MSI for jobs - I would add a comment about it

File: samples/src/main/python/SynapsePythonSDK.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# This sample shows how to use the Kusto Python SDK with Synapse authentication.
+# One should use the SDK to run admin commands while using the connector to query
+# https://docs.microsoft.com/azure/data-explorer/kusto/api/python/kusto-python-client-library
+# To upload SDK package to synapse workspace, create a wheel file and follow this:
+# https://docs.microsoft.com/azure/synapse-analytics/spark/apache-spark-manage-python-packages#install-wheel-files
+from azure.kusto.data import KustoClient, KustoConnectionStringBuilder
+
```
Comment (noam-shapira) [inline samples/src/main/python/SynapsePythonSDK.py:7]: I would add an example with LS credentials, as that what advance users are looking for

File: samples/src/main/python/SynapsePythonSDK.py (hunk 1 above)
Comment (ohadbitt) [reply samples/src/main/python/SynapsePythonSDK.py:8]: Can you add a suggestion comment with how you want to explain it 
Or add tge other way if you think it is relevant

**Review by noam-shapira — APPROVED** [review]

File: samples/src/main/python/SynapsePythonSDK.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# This sample shows how to use the Kusto Python SDK with Synapse Spark utilities for authentication as described here:
+# https://docs.microsoft.com/azure/synapse-analytics/spark/microsoft-spark-utilities?pivots=programming-language-python#credentials-utilities.
+# One should use the SDK to run admin commands while using the connector to query (as in SynapseSample.py)
+# https://docs.microsoft.com/azure/data-explorer/kusto/api/python/kusto-python-client-library
+# To upload SDK package to synapse workspace, create a wheel file (if only the Python SDK is needed - clone this repo:
+# https://github.com/Azure/azure-kusto-python) and then follow this:
+# https://docs.microsoft.com/azure/synapse-analytics/spark/apache-spark-manage-python-packages#install-wheel-files
+# In this sample the principal should have at least database monitor privileges.
```
Comment (yihezkel) [inline samples/src/main/python/SynapsePythonSDK.py:8]:
```suggestion
# In this sample, the principal should have at least database monitor privileges.
```

File: samples/src/main/python/SynapsePythonSDK.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+from azure.kusto.data import KustoClient, KustoConnectionStringBuilder
+
+# A token provider that uses the linked service authentication method provided in its creation.
+token_provider = lambda: mssparkutils.credentials.getConnectionStringOrCreds("{linked_service_name}")
+
+# A token provider that uses the user's credentials in interactive mode via MSI.
```
Comment (yihezkel) [inline samples/src/main/python/SynapsePythonSDK.py:14]:
```suggestion
# A token provider that uses the user's credentials in interactive mode via managed identity.
```

**Review by yihezkel — APPROVED** [review]
A few minor grammar suggestions

**Pull request merged**